spec/builtins2.test.sh

```diff
@@ -136,3 +136,38 @@ status=127
 status=127
 status=127
 ## END
+
+#### builtin
+cd () { echo "hi"; }
+cd
+builtin cd / && pwd
+unset -f cd
+## STDOUT:
+hi
+/
+## N-I dash STDOUT:
+hi
+## N-I dash STDERR:
+/bin/dash: 3: builtin: not found
+
+#### builtin ls not found
+builtin ls
+## status: 1
+## OK bash STDERR:
+/bin/bash: line 1: builtin: ls: not a shell builtin
```

Comment:
It looks like there need to be a bunch of `## END` here:

Is the rest green or yellow for all shells?   Mine looks like this:

```
andy@lisa:~/git/oilshell/oil$ test/spec.sh builtins2
builtins2.test.sh
case    line    dash    bash    mksh    zsh     osh
  0       3     ok      pass    pass    pass    pass    command -v
  1      31     BUG     BUG     ok      pass    pass    command -v with multiple names
  2      60     pass    pass    pass    pass    pass    command skips function lookup
  3      80     pass    pass    pass    N-I     pass    command command seq 3
  4      90     pass    pass    pass    N-I     pass    command command -v seq
  5      99     N-I     ok      BUG     BUG     pass    history usage

```

Reply:
Yes, they are all `pass`, `ok`, or `N-I`. I don't mind adding `## END` but all the tests were working without it.

Is there a way to have the tests output `pass` for an output specific to a shell? When I add the following I get a parse error:

```
 ## status: 1
-## OK bash STDERR:
+## PASS bash STDERR:
```

```
Traceback (most recent call last):
  File "test/sh_spec.py", line 996, in <module>
    sys.exit(main(sys.argv))
  File "test/sh_spec.py", line 915, in main
    cases = ParseTestFile(tokens)
  File "test/sh_spec.py", line 310, in ParseTestFile
    test_case = ParseTestCase(tokens)
  File "test/sh_spec.py", line 279, in ParseTestCase
    assert kind == TEST_CASE_BEGIN, (line_num, kind, item)  # Invariant
AssertionError: (157, 4, '/bin/bash: line 1: builtin: ls: not a shell builtin')
```

Reply:
It's not supposed to be green/ PASS unless the output is exactly what's expected.  It can be yellow as BUG or N-I though; that's the intended usage.

I don't think it could have ever worked without ## END ?  In any case it looks right now.